source/sgp_mode/Instructions.h

```diff
@@ -124,7 +124,8 @@ INST(SharedIO, {
   state.input_buf.push(next);
 });
 INST(Donate, {
-  if (state.host->IsHost())
+  if (state.world->GetConfig()->DONATION()){
+      if (state.host->IsHost())
     return;
```

Comment:
Please improve the indentation here so that it's clear what is and isn't within your new if block